src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1473,6 +1474,63 @@ private static bool DatabaseNeedsToRunIdleOperations(DocumentDatabase database)
             return ((now - maxLastWork).TotalMinutes > 5) || ((now - database.LastIdleTime).TotalMinutes > 10);
         }
 
+        public void AssignNodesToDatabase(ClusterTopology clusterTopology, DatabaseRecord record)
+        {
+            var topology = record.Topology;
+
+            Debug.Assert(topology != null);
+
+            var clusterNodes = clusterTopology.Members.Keys
+                .Concat(clusterTopology.Promotables.Keys)
+                .Concat(clusterTopology.Watchers.Keys)
+                .ToList();
+
+            if (record.Encrypted)
+            {
+                clusterNodes.RemoveAll(n => AdminDatabasesHandler.NotUsingHttps(clusterTopology.GetUrlFromTag(n)));
+                if (clusterNodes.Count == 0)
```

Comment:
Need to compare to the replication factor here, no?

Reply:
OK, done.